### PR TITLE
fix: upgrade pdfjs to 2.6.347

### PR DESCRIPTION
**Pull request by ColbyJohnIBM**
#### What do these changes do/fix?

- upgrade pdfjs-dist version in package.json
- update types according to new types system (internally build rather than DefinitelyTyped)

Contributes to https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/12076

#### How do you test/verify these changes?
- Run storybook and check that PDF stories work correctly.
- Link with tooling and ensure that PDFs render correctly. Note that the linked issue is fixed.

### CODE REVIEW

**Comment by ColbyJohnIBM** [comment]
Unit tests fail with this error:

```
Summary of all failing tests
FAIL src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx (15.018 s)
  ● PdfViewer › renders PDF document



      15 | type RenderTask = ReturnType<PDFPageProxy['render']>;
      16 |
    > 17 | const PdfjsLib = require('pdfjs-dist');
         |                  ^
      18 |
      19 | setupPdfjs();
      20 |

      at BaseExceptionClosure (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/src/shared/util.js:428:29)
      at Object.<anonymous> (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/src/shared/util.js:419:24)
      at __w_pdfjs_require__ (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/bootstrap:19:1)
      at Object.<anonymous> (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/src/display/display_utils.js:17:1)
      at __w_pdfjs_require__ (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/bootstrap:19:1)
      at Object.<anonymous> (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/src/pdf.js:16:1)
      at __w_pdfjs_require__ (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/bootstrap:19:1)
      at ../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/bootstrap:83:1
      at ../../node_modules/pdfjs-dist/build/pdf.js:279:5
      at webpackUniversalModuleDefinition (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/universalModuleDefinition:3:1)
      at Object.<anonymous> (../../node_modules/pdfjs-dist/build/webpack:/pdfjs-dist/build/pdf/webpack/universalModuleDefinition:1:1)
      at Object.<anonymous> (src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx:17:18)
      at Object.<anonymous> (src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:3:1)
```

The issue seems related to the way that things are being exported at a top level from version `2.4.456` to `2.5.207`: https://github.com/mozilla/pdfjs-dist/compare/v2.4.456...v2.5.207#diff-a5300ddf3b50734db4f5951cc53e5b86a26644eb7bc791f78d2fbf5161929f30

I'm not sure how to get around it yet, but I'll keep digging on that on Monday.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { FC, useEffect, useRef, useCallback, useMemo, useState } from 'react';
 import cx from 'classnames';
-import PdfjsLib, { PDFDocumentProxy, PDFPageProxy, PDFPromise, PDFRenderTask } from 'pdfjs-dist';
+import * as PdfjsLib from 'pdfjs-dist';
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx:3]: Seems that we only use var `PdfjsLib` for `getDocument()` and `GlobalWorkerOptions`. Why not just bring those in, instead of `import *`?
```suggestion
import { getDocument, GlobalWorkerOptions } from 'pdfjs-dist';
```

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/typings.d.ts [hunk 2]
```diff
@@ -15,20 +15,21 @@ declare module 'pdfjs-dist/lib/web/ui_utils' {
 
 declare module 'pdfjs-dist/lib/web/text_layer_builder' {
   import { EventBus } from 'pdfjs-dist/lib/web/ui_utils';
-  import PdfjsLib from 'pdfjs-dist';
+  import { PageViewport } from 'pdfjs-dist/types/display/display_utils';
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/typings.d.ts:18]: Is this file still needed? Or can we just remove this and get what we need from upgraded library's typings?

**Comment by jhpedemonte** [comment]
Seeing this during build:

```
(!) Use of eval is strongly discouraged
https://rollupjs.org/guide/en/#avoiding-eval
../../node_modules/pdfjs-dist/build/pdf.js
2895: 
2896:       if (_is_node.isNodeJS && typeof require === "function") {
2897:         const worker = eval("require")(getWorkerSrc());
                             ^
2898:         return worker.WorkerMessageHandler;
2899:       }
../../node_modules/pdfjs-dist/lib/display/api.js
1149: 
1150:       if (_is_node.isNodeJS && typeof require === "function") {
1151:         const worker = eval("require")(getWorkerSrc());
                             ^
1152:         return worker.WorkerMessageHandler;
1153:       }
```

We should investigate if this is correct, or if we're building this incorrectly, somehow.

**Comment by ColbyJohnIBM** [comment]
So strange, `PdfViewer.test.tsx` passes locally and fails in the build... 😖 

```
➤ YN0000: [@ibm-watson/discovery-react-components]: FAIL src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx (8.833 s)
➤ YN0000: [@ibm-watson/discovery-react-components]:   ● PdfViewer › renders PDF document
➤ YN0000: [@ibm-watson/discovery-react-components]: 
➤ YN0000: [@ibm-watson/discovery-react-components]:     thrown: "Exceeded timeout of 5000 ms for a test.
➤ YN0000: [@ibm-watson/discovery-react-components]:     Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
➤ YN0000: [@ibm-watson/discovery-react-components]: 
➤ YN0000: [@ibm-watson/discovery-react-components]:        7 |
➤ YN0000: [@ibm-watson/discovery-react-components]:        8 | describe('PdfViewer', () => {
➤ YN0000: [@ibm-watson/discovery-react-components]:     >  9 |   it('renders PDF document', async () => {
➤ YN0000: [@ibm-watson/discovery-react-components]:          |   ^
➤ YN0000: [@ibm-watson/discovery-react-components]:       10 |     render(<PdfViewer file={atob(doc)} page={1} scale={1} setLoading={(): void => {}} />);
➤ YN0000: [@ibm-watson/discovery-react-components]:       11 |
➤ YN0000: [@ibm-watson/discovery-react-components]:       12 |     // wait for component to finish rendering (prevent "act" warning)
➤ YN0000: [@ibm-watson/discovery-react-components]: 
➤ YN0000: [@ibm-watson/discovery-react-components]:       at src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:9:3
➤ YN0000: [@ibm-watson/discovery-react-components]:       at Object.<anonymous> (src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:8:1)
```

**Review by jhpedemonte — APPROVED** [review]
